### PR TITLE
re-add sentry

### PR DESCRIPTION
**Pull request by lsmith77**
We had Sentry before which I removed because I didn't want to have them in our processors list as their hosting is in the US however we will eventually self-host

### CODE REVIEW

File: app/sentry.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import sentry_sdk
+from sentry_sdk.integrations.aiohttp import AioHttpIntegration
+from posthog.sentry.posthog_integration import PostHogIntegration
+
+# Sentry SDK set up with PostHog integration
+def set_up_sentry_sdk(version, settings):
+    if not settings.sentry_dsn or settings.testing == True:
+        return False
```
Comment (Szarad) [inline app/sentry.py]: this line is causing the problem, sentry_sdk is set to False during testing

**Review by Szarad — CHANGES_REQUESTED** [review]

File: app/sentry.py (hunk 1 above)
Comment (Szarad) [reply app/sentry.py]: We can here `return None` and check the value in main.py in line 439

File: app/main.py [hunk 2]
```diff
@@ -409,6 +433,11 @@ def log(
 async def check_query(
     request: Request, user_request_in: RequestIn, background_tasks: BackgroundTasks
 ):
+    if not user_request_in.id == None:
+        posthog.identify(id)
+
+    sentry_sdk.set_context("request", clean_requestin_data(user_request_in))
```
Comment (Szarad) [inline app/main.py]:
```python
if sentry_sdk: 
    sentry_sdk.set_context("request", clean_requestin_data(user_request_in))
 ```

**Review by Szarad — CHANGES_REQUESTED** [review]

File: app/sentry.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+from posthog.sentry.posthog_integration import PostHogIntegration
+
+# Sentry SDK set up with PostHog integration
+def set_up_sentry_sdk(version, settings):
+    if not settings.sentry_dsn or settings.testing == True:
+        return None
+
+    integrations = [AioHttpIntegration()]
+
+    if settings.training_data_enabled:
```
Comment (lsmith77) [inline app/sentry.py]: note right now I have not been able to get the posthog integration to work :-/

Comment (lsmith77) [reply app/sentry.py]: see also https://github.com/PostHog/posthog-python/issues/46

**Review by Szarad — APPROVED** [review]
Looks good to me

**Pull request merged**